cassiopeia/plotting/local.py

```diff
@@ -0,0 +1,829 @@
+    loc: Literal["left", "right", "up", "down", "polar"],
+) -> float:
+    """Compute the size of colorstrip boxes.
+
+    This function computes two sizes: the width and height of the colorstrip
+    boxes. The height is selected to be the value such that the boxes are
+    tightly arranged next to one another. The width is selected to be 5% of the
+    tree depth.
+
+    Args:
+        node_coords: Node coordinates.
+        anchor_coords: Anchor coordinates.
+        loc: The location of the box relative to the anchor.
+
+    Returns:
+        Two floats representing the width and height of the box.
+    """
+    min_pos = np.inf
+    max_pos = -np.inf
+    min_depth = np.inf
+    max_depth = -np.inf
+    for x, y in node_coords.values():
+        pos, depth = x, y
+        if loc in ("left", "right"):
+            pos, depth = y, x
+        min_pos = min(min_pos, pos)
+        max_pos = max(max_pos, pos)
+        min_depth = min(min_depth, depth)
+        max_depth = max(max_depth, depth)
+    return (
+        (max_depth - min_depth) * 0.05,
+        (max_pos - min_pos + 1) / len(anchor_coords),
+    )
+
+
+def create_categorical_colorstrip(
+    values: Dict[str, str],
+    anchor_coords: Dict[str, Tuple[float, float]],
+    width: float,
+    height: float,
+    spacing: float,
+    loc: Literal["left", "right", "up", "down", "polar"],
+    cmap: Union[str, mpl.colors.Colormap] = "tab10",
+) -> Tuple[
+    Dict[str, Tuple[List[float], List[float], Tuple[float, float, float], str]],
+    Dict[str, Tuple[float, float]],
+]:
+    """Create a colorstrip for a categorical variable.
+
+    Args:
+        values: Dictionary containing node-category pairs.
+        anchor_coords: Anchor coordinates for the colorstrip boxes.
+        width: Width of the colorstrip.
+        height: Height of each box of the colorstrip.
+        spacing: The amount of space to "skip" before placing each box
+            (i.e. padding)
+        loc: Location of the boxes relative to the anchors.
+        cmap: Colormap. Defaults to the `tab10` colormap of Matplotlib.
+
+    Returns:
+        Dictionary of box coordinates and a dictionary of new anchor coordinates.
+    """
+    cm = plt.cm.get_cmap(cmap)
+    unique_values = set(values.values())
+    value_mapping = {val: i for i, val in enumerate(unique_values)}
+
+    boxes, next_anchor_coords = utilities.place_colorstrip(
+        anchor_coords, width, height, spacing, loc
+    )
+    colorstrip = {}
+    for leaf, val in values.items():
+        v = value_mapping[val] % cm.N
+        colorstrip[leaf] = boxes[leaf] + (cm(v)[:-1], f"{leaf}\n{val}")
+    return colorstrip, next_anchor_coords
+
+
+def create_continuous_colorstrip(
+    values: Dict[str, float],
+    anchor_coords: Dict[str, Tuple[float, float]],
+    width: float,
+    height: float,
+    spacing: float,
+    loc: Literal["left", "right", "up", "down", "polar"],
+    cmap: Union[str, mpl.colors.Colormap] = "viridis",
```

Comment:
Is it worthwhile to support adding `vmin` or `vmax` parameters (perhaps percentiles, like `scanpy` supports?)

Reply:
Yep, although I don't think its worth adding support for the percentile string...

Reply:
For sure, that's fine. I definitely make use of the percentile strings in `scanpy`, but perhaps I'm the only one...